calendar/tests/test.bal

```diff
@@ -119,14 +124,18 @@ function testquickAdd() {
 }
 function testGetEvents() {
     log:printInfo("calendarClient -> getEvents()");
-    stream<Event,error> resultStream = calendarClient->getEvents(testCalendarId);
-    record {|Event value;|}|error? res = resultStream.next();
-    if (res is record {|Event value;|}) {
-        test:assertNotEquals(res.value["id"], "", msg = "Found 0 records");
-    } else {
-        if(res is error){
-            test:assertFail(res.message());
+    stream<Event,error>|error resultStream = calendarClient->getEvents(testCalendarId);
+    if (resultStream is stream<Event,error>) {
+        record {|Event value;|}|error? res = resultStream.next();
+        if (res is record {|Event value;|}) {
+            test:assertNotEquals(res.value["id"], "", msg = "Found 0 records");
```

Comment:
@MaryamZi could you please verify if this approach is the best to check if stream is empty. @Chuhaa I believe that is what you try to check in here?

Reply:
Stream emptiness is indicated by `res` not being the record, and being a subtype of the completion type (error in this case). That would be the else block at L132.

Also, since the completion type of this stream doesn't contain nil, isn't res's type `record {|Event value;|}|error` (without nil)? By using the exact type you can get rid of the check in L133, since `res` will be an error in the else block.

If it actually could contain nil, I guess we should explicitly handle the case where `res` is nil, but currently in the else block we are calling `assertFail` only if `res` is an error?

Reply:
Yes. Response could contain nil as well and the current test doesn't cover it.  

@MaryamZi Could you please review whether following is a correct approach for L130 to L136?
```ballerina
if (res is error) {
      test:assertFail(res.message());
} else {
      test:assertTrue(res is record {|Event value;|}, msg = "Found 0 records");
}
```

Reply:
Since `resultStream` is `stream<Event, error>` if it is a stream, it will only return either the record or an error when calling `next()`. Nil is returned only when the completion type has nil (e.g., `stream<Event, error?>`, `stream<Event>`, etc).

So you can probably do

```ballerina
record {|Event value;|}|error res = resultStream.next();
```

`res is record {|Event value;|}` in the else block will then always evaluate to true. So you can maybe assert the value if needed or replace the if-else block with

```ballerina
test:assertTrue(res is record {|Event value;|}, msg = "Found 0 records");
```

Reply:
Yes we can. Thanks @MaryamZi